janusgraph-cql/src/main/java/org/janusgraph/diskstorage/cql/CQLStoreManager.java

```diff
@@ -0,0 +1,421 @@
+                                        .grouped(2)
+                                        .toMap(array -> Tuple.of(array.get(0), Integer.parseInt(array.get(1)))))))
+                .toJavaMap();
+
+        final Session s = this.cluster.connect();
+        s.execute(createKeyspace(keyspaceName)
+                .ifNotExists()
+                .with()
+                .replication(replication));
+        return s;
+    }
+
+    ExecutorService getExecutorService() {
+        return this.executorService;
+    }
+
+    Session getSession() {
+        return this.session;
+    }
+
+    String getKeyspaceName() {
+        return this.keyspace;
+    }
+
+    Map<String, String> getCompressionOptions(final String name) throws BackendException {
+        final KeyspaceMetadata keyspaceMetadata = Option.of(this.cluster.getMetadata().getKeyspace(this.keyspace))
+                .getOrElseThrow(() -> new PermanentBackendException(String.format("Unknown keyspace '%s'", this.keyspace)));
+        return Option.of(keyspaceMetadata.getTable(name))
+                .map(tableMetadata -> tableMetadata.getOptions().getCompression())
+                .getOrElseThrow(() -> new PermanentBackendException(String.format("Unknown table '%s'", name)));
+    }
+
+    @Override
+    public void close() throws BackendException {
+        try {
+            this.session.close();
+        } finally {
+            try {
+                this.cluster.close();
+            } finally {
+                this.executorService.shutdownNow();
+            }
+        }
+    }
+
+    @Override
+    public String getName() {
+        return String.format("%s.%s", getClass().getSimpleName(), this.keyspace);
+    }
+
+    @Override
+    public Deployment getDeployment() {
+        return this.deployment;
+    }
+
+    @Override
+    public StoreFeatures getFeatures() {
+        return this.storeFeatures;
+    }
+
+    @Override
+    public KeyColumnValueStore openDatabase(final String name, final Container metaData) throws BackendException {
+        return this.openStores.computeIfAbsent(name, n -> new CQLKeyColumnValueStore(this, n, getStorageConfig(), () -> this.openStores.remove(n)));
+    }
+
+    @Override
+    public StoreTransaction beginTransaction(final BaseTransactionConfig config) throws BackendException {
+        return new CQLTransaction(config);
+    }
+
+    @Override
+    public void clearStorage() throws BackendException {
+        final Future<Seq<ResultSet>> result = Future.sequence(
+                this.executorService,
+                Iterator.ofAll(this.cluster.getMetadata().getKeyspace(this.keyspace).getTables())
+                        .map(table -> Future.fromJavaFuture(this.executorService, this.session.executeAsync(truncate(this.keyspace, table.getName())))));
+        result.await();
+        if (result.isFailure()) {
+            throw EXCEPTION_MAPPER.apply(result.getCause().get());
+        }
```

Comment:
Since with dropKeyspace all tests pass consistently what about adding it back for now and creating a separate issue for updating to start a clean embedded instance for each test class and/or finding a dropKeyspace alternative so we can break that work off from this PR and move forward here?

The dropKeyspace usage doesn't seem to have a problem with hanging and it's called multiple times for each of over 2000 tests in the TinkerPop suite alone, so it seems like if it was going to hang it would at some point there. Also I haven't observed the intermittent failures due to connection startup (beyond the keyspace issue) that you mention. Can you elaborate on what you're seeing there?